src/app/domain/backup/services/uploadMedias.ts

```diff
@@ -0,0 +1,196 @@
+/* eslint-disable promise/always-return */
+import Minilog from '@cozy/minilog'
+import { CameraRoll } from '@react-native-camera-roll/camera-roll'
+import { Platform } from 'react-native'
+import RNBackgroundUpload, {
+  UploadOptions
+} from 'react-native-background-upload'
+import RNFileSystem from 'react-native-fs'
+
+import { setMediaAsBackuped } from '/app/domain/backup/services/manageLocalBackupConfig'
+import { Media } from '/app/domain/backup/models/Media'
+
+import type CozyClient from 'cozy-client'
+
+const log = Minilog('💿 Backup')
+
+interface UploadMediaResult {
+  success: boolean
+  data?: object
+}
+
+export const uploadMedias = async (
+  client: CozyClient,
+  backupFolderId: string,
+  mediasToUpload: Media[]
+): Promise<boolean> => {
+  for (const mediaToUpload of mediasToUpload) {
+    try {
+      const uploadUrl = getDriveUrl(client, backupFolderId, mediaToUpload)
+
+      const { success } = await uploadMedia(client, uploadUrl, mediaToUpload)
+
+      if (success) {
+        log.debug(`✅ ${mediaToUpload.name} uploaded`)
+        await setMediaAsBackuped(client, mediaToUpload)
```

Comment:
There is a lot of stuff done in `setMediaAsBackuped`: 
- We read from the localStorage 
- We map on all the backupedMedia array 
- We add the file to an array 
- We map on another array to remove the file to sync 
- We finally save the data to the localStorage

And we do that for every media we upload. Do we have an idea of the time taken just by this method during the first backup? 

At least the code and the path is readable and understandable, I just want to be sure that it'll not slow down too much the process

Reply:
**Quick test**

Around 10ms per call on an iPhone SE 2020 with 10 files.
Around 20ms per call on an iPhone SE 2020 with 200 files.

So localStorage is not an issue but the multiple map could be with a lot of medias.

Reply:
Great. Let's go that way and see how it performs later?

Reply:
Yes! I will create a task to check performance with 1000+ and 5000+ medias and stay vigilant about big arrays. I'm also relieved about local storage now because I use it extensively...